### ISSUE TITLE
Addition of multiple Frontend IP configuration, LB rules, & Health probes to exiting LB  un missing

### ISSUE THREAD

**Issue by vikram-chandrakanth** [hashicorp/terraform-provider-azurerm#2424]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.11.7

### Affected Resource(s)

* azurerm_lb"

### Terraform Configuration Files

Hi Team, 

As per the terraform documentation for the azurerm providers, there seems to be the limitation of the creation of the front end ip configuration per azure lb. i am looking for additional of the mulitple fronend-ip configuration , lb rules, health probes to the already existing loadbalancer in azure. 

as the work around, i has written multiple block of the front end IP in the azure lb block, but it keeps  interchanging  the IP's  every time we run transform. 

```hcl

resource "azurerm_lb" "lb" {
  resource_group_name = "${var.elb_rg}"
  name                = "${var.lb_name}"
  location            = "${var.location}"
  sku = "Standard"

frontend_ip_configuration {
    name = "${azurerm_public_ip.lb_fend.name}"
    public_ip_address_id = "${azurerm_public_ip.lb_fend.id}"
  }


 frontend_ip_configuration {
    name = "${replace("${azurerm_public_ip.lb_pip.*.name[0]}","pip", "vip")}"
    public_ip_address_id = "${element(azurerm_public_ip.lb_pip.*.id, 0)}"
  }

 frontend_ip_configuration {
    name = "${replace("${azurerm_public_ip.lb_pip.*.name[1]}","pip", "vip")}"
    public_ip_address_id = "${element(azurerm_public_ip.lb_pip.*.id, 1)}"
  }
}

resource "azurerm_lb_rule" "create_rules" {
 depends_on = ["azurerm_lb_probe.health_probe"]
 count = "${length(var.vips)}"
  resource_group_name            = "${var.elb_rg}"
  loadbalancer_id                = "${azurerm_lb.lb.id}"
  name                           = "${lookup(var.vips[count.index],"name")}-rule"
  protocol                       = "${lookup(var.vips[count.index],"rule_protocol")}"
  frontend_port                  = "${lookup(var.vips[count.index], "frontend_port")}"
  backend_port                   = "${lookup(var.vips[count.index], "probe_port")}"
  frontend_ip_configuration_name = "${lookup(var.vips[count.index],"name")}-vip"
  #backend_address_pool_id  = "${azurerm_lb_backend_address_pool.backend_pool.id}"
  probe_id  = "${element(azurerm_lb_probe.health_probe.*.id, count.index)}"
}


resource "azurerm_lb_probe" "health_probe" {
  count = "${length(var.vips)}"
  resource_group_name = "${var.elb_rg}"
  loadbalancer_id     = "${azurerm_lb.lb.id}"
  name                = "F5Probe-${lookup(var.vips[count.index], "probe_port")}"
  port                = "${lookup(var.vips[count.index], "probe_port")}"
}


```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by sampcoug** [comment]
How do I go about adding multiple IPs in the azurerm_lb_rule"? The VMs are behind a load balancer and I want to have port 80 and port 22 access to the VMs. I can only add one "frontend_port" at a time. I tried adding port 22 as a nat rule - "azurerm_lb_nat_rule" and that didn't seem to work. 
Any idea folks? 

`resource "azurerm_lb_rule" "lb_rule" {
  resource_group_name            = "${azurerm_resource_group.rg.name}"
  loadbalancer_id                = "${azurerm_lb.lb.id}"
  name                           = "LBRule"
  protocol                       = "tcp"
  frontend_port                  = **80, 22**
  backend_port                   = **80, 22**
  frontend_ip_configuration_name = "LoadBalancerFrontEnd"
  enable_floating_ip             = false
  backend_address_pool_id        = "${azurerm_lb_backend_address_pool.backend_pool.id}"
  idle_timeout_in_minutes        = 5
  probe_id                       = "${azurerm_lb_probe.lb_probe.id}"
  depends_on                     = ["azurerm_lb_probe.lb_probe"]
}`

**Comment by Hkyx** [comment]
same problem, same code + -. seems you can't iterate accross your lb_probe values.

**Comment by warrenackerman** [comment]
I need this as well. I was initially trying a similar bit of code and when it failed a google search brought me here. This is a big gap for us because there is no work around. You cannot create multiple LBs and map to the same back end. Azure will not allow it.

**Comment by porwalameet** [comment]
Yes..even i am blocked on this. I want to attach a new IP to existing loadbalancer. 
My Setup is like this:
AKS Cluster in azure which create a agentpool RG and a service of  LoadBalancer named " kubernetes". Now if i want to attach a new public IP to this loadbalancer server, it is not allowing me to do that. Any update or workaround on this?
The reason i want to attached to this LB is cos I need same set of nodepools as backend.

**Comment by EvilSmiff** [comment]
Blocked for a number of things by the frontend ip configuration.  It makes the azurerm_lb is pretty useless for our case.  We use the load balancer for active passive clusters with a flexible number of frontend ip configurations, rules and probes.  The limitations mean this is a core piece of infrastructure we just cannot deploy with terraform.  We do this with both internal and external load balancers.

**Comment by favoretti** [comment]
@EvilSmiff can you please elaborate on this one? adding multiple `frontend_ip_configuration` is definitely supported. Would you be able to open a fresh new issue describing the problem you're running into? Since this issue has been reported a long time ago and relates to the version of provider we no longer support - I'm going to close it. Looking forward to working with you on your challenges in a new report. Thank you.
